tests/test_classifiers.py

```diff
@@ -0,0 +1,265 @@
+# limitations under the License.
+
+import os
+
+import pytest
+from distributed import Client
+
+from nemo_curator import get_client
+from nemo_curator.datasets import DocumentDataset
+from nemo_curator.utils.import_utils import gpu_only_import, gpu_only_import_from
+
+cudf = gpu_only_import("cudf")
+dask_cudf = gpu_only_import("dask_cudf")
+LocalCUDACluster = gpu_only_import_from("dask_cuda", "LocalCUDACluster")
+
+
+@pytest.fixture
+def gpu_client(request):
+    with LocalCUDACluster(n_workers=1) as cluster, Client(cluster) as client:
+        request.client = client
+        request.cluster = cluster
+        yield
+
+
+@pytest.fixture
+def domain_dataset():
+    text = [
+        "Quantum computing is set to revolutionize the field of cryptography.",
+        "Investing in index funds is a popular strategy for long-term financial growth.",
+        "Recent advancements in gene therapy offer new hope for treating genetic disorders.",
+        "Online learning platforms have transformed the way students access educational resources.",
+        "Traveling to Europe during the off-season can be a more budget-friendly option.",
+    ]
+    df = cudf.DataFrame({"text": text})
+    df = dask_cudf.from_cudf(df, 1)
+    return DocumentDataset(df)
+
+
+@pytest.mark.gpu
+def test_domain_classifier(gpu_client, domain_dataset):
+    from nemo_curator.classifiers import DomainClassifier
+
+    classifier = DomainClassifier()
+    result_dataset = classifier(dataset=domain_dataset)
+    result_pred = result_dataset.df.compute()["domain_pred"]
+
+    expected_pred = cudf.Series(
+        [
+            "Computers_and_Electronics",
+            "Finance",
+            "Health",
+            "Jobs_and_Education",
+            "Travel_and_Transportation",
+        ]
+    )
+
+    assert result_pred.equals(expected_pred)
+
+
+@pytest.mark.gpu
+def test_quality_classifier(gpu_client):
+    from nemo_curator.classifiers import QualityClassifier
+
+    text = [
+        "Traveling to Europe during the off-season can be a more budget-friendly option."
+    ]
+    df = cudf.DataFrame({"text": text})
+    input_dataset = DocumentDataset(dask_cudf.from_cudf(df, npartitions=1))
+
+    classifier = QualityClassifier()
+    result_dataset = classifier(dataset=input_dataset)
+    result_pred = result_dataset.df.compute()["quality_pred"]
+
+    expected_pred = cudf.Series(["Medium"])
+
+    assert result_pred.equals(expected_pred)
+
+
+@pytest.mark.gpu
+@pytest.mark.parametrize(
+    "aegis_variant",
+    [
+        "nvidia/Aegis-AI-Content-Safety-LlamaGuard-Defensive-1.0",
+        "nvidia/Aegis-AI-Content-Safety-LlamaGuard-Permissive-1.0",
+    ],
+)
+def test_aegis_classifier(gpu_client, aegis_variant):
+    from nemo_curator.classifiers import AegisClassifier
+
+    hugging_face_api_token = os.getenv("HUGGING_FACE_API_TOKEN")
+    if not hugging_face_api_token:
+        pytest.fail("HUGGING_FACE_API_TOKEN environment variable not set")
```

Comment:
We grab the Hugging Face token here instead of using a `@pytest.fixture`. The reason for this is that if we used a PyTest fixture and the test failed for whatever reason, the Hugging Face token would be exposed in the logs.

Reply:
If there is not a short term fix here, we can probably skip this test using `skipif` . 

```
@pytest.mark.skipif(
    not os.getenv("HUGGING_FACE_API_TOKEN"),
    reason="HUGGING_FACE_API_TOKEN environment variable is not set"
)
```